transport/transport.go

```diff
@@ -341,8 +341,10 @@ func NewServerTransport(protocol string, conn net.Conn, maxStreams uint32, authI
 type ConnectOptions struct {
 	// UserAgent is the application user agent.
 	UserAgent string
-	// Dialer specifies how to dial a network address.
-	Dialer func(string, time.Duration) (net.Conn, error)
+	// Cancel is closed to indicate that dialing should be cancelled.
+	Cancel chan struct{}
+	// dialer specifies how to dial a network address.
+	dialer func(string, time.Duration, <-chan struct{}) (net.Conn, error)
```

Comment:
Any reason this is not exported? All the other fields are exported. If it is exported, it seems we do not need ConnectOptions.WithDialer(...).